azure/converters/spotinstances.go

```diff
@@ -40,9 +40,9 @@ func GetSpotVMOptions(spotVMOptions *infrav1.SpotVMOptions, diffDiskSettings *in
 			MaxPrice: &maxPrice,
 		}
 	}
-	evictionPolicy := compute.VirtualMachineEvictionPolicyTypesDeallocate
-	if diffDiskSettings != nil && diffDiskSettings.Option == "Local" {
-		evictionPolicy = compute.VirtualMachineEvictionPolicyTypesDelete
-	}
+
+	// Set the spot vm eviction policy if provided.
+	evictionPolicy := compute.VirtualMachineEvictionPolicyTypes(*spotVMOptions.EvictionPolicy)
```

Comment:
we want to make sure we guarding against any potential nil pointer dereferencing here (even though we don't expect this to ever be nil in practice)

```
// Set the spot vm eviction policy if provided.
if spotVMOptions != nil {
    evictionPolicy := compute.VirtualMachineEvictionPolicyTypes(*spotVMOptions.EvictionPolicy)
}
```


This will also fix the unit test failures you're seeing along the lines of `panic: runtime error: invalid memory address or nil pointer dereference`

Reply:
This makes sense! I implemented this, but the unit tests that test if the spot vm converter returns a default eviction policy is now broken, since the defaults are being set in the webhook. Should I just remove the [test](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2a0afb510a3554f4aacfaa411d9c66cd2245ecbe/azure/services/virtualmachines/spec_test.go#L273) entirely?